src/settings_table.cpp

```diff
@@ -111,9 +111,7 @@ static void StationSpreadChanged(int32 p1)
 
 static void CloseSignalGUI(int32 new_value)
 {
-	if (new_value == 0) {
-		CloseWindowByClass(WC_BUILD_SIGNAL);
-	}
+	CloseWindowByClass(WC_BUILD_SIGNAL);
 }
```

Comment:
Settings callbacks can be lambdas now (see `gui.population_in_label`), can this function be removed entirely and just call `CloseWindowByClass` directly ?